sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1495,6 +1495,56 @@ pi_result piDevicesGet(pi_platform Platform, pi_device_type DeviceType,
   return PI_SUCCESS;
 }
 
+// sub-sub-device
+pi_result getComputeCmdQueueOrdinals(
+    pi_device PiSubDevice, std::vector<int> &Ordinals,
+    std::vector<ze_command_queue_group_properties_t> &AllQueueProperties) {
+  uint32_t numQueueGroups = 0;
+  ZE_CALL(zeDeviceGetCommandQueueGroupProperties,
+          (PiSubDevice->ZeDevice, &numQueueGroups, nullptr));
+  if (numQueueGroups == 0) {
+    return PI_ERROR_UNKNOWN;
+  }
+  std::vector<ze_command_queue_group_properties_t> QueueProperties(
+      numQueueGroups);
+  ZE_CALL(zeDeviceGetCommandQueueGroupProperties,
+          (PiSubDevice->ZeDevice, &numQueueGroups, QueueProperties.data()));
+
+  bool noComputeEngineFlag = true;
+  for (uint32_t i = 0; i < numQueueGroups; i++) {
+    if (QueueProperties[i].flags &
+        ZE_COMMAND_QUEUE_GROUP_PROPERTY_FLAG_COMPUTE) {
+      Ordinals.push_back(i);
+      AllQueueProperties.push_back(QueueProperties[i]);
+      noComputeEngineFlag = false;
+    }
+  }
+
+  if (noComputeEngineFlag) {
```

Comment:
you can save this variable by just checking if AllQueueProperties.empty()

Reply:
right

Reply:
done.